core/src/main/java/org/infinispan/configuration/parsing/FragmentParser.java

```diff
@@ -0,0 +1,1911 @@
+import org.infinispan.commons.util.Util;
+import org.infinispan.configuration.cache.AbstractStoreConfigurationBuilder;
+import org.infinispan.configuration.cache.AsyncStoreConfigurationBuilder;
+import org.infinispan.configuration.cache.AuthorizationConfigurationBuilder;
+import org.infinispan.configuration.cache.BackupConfiguration;
+import org.infinispan.configuration.cache.BackupConfigurationBuilder;
+import org.infinispan.configuration.cache.BackupFailurePolicy;
+import org.infinispan.configuration.cache.BiasAcquisition;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ClusterLoaderConfigurationBuilder;
+import org.infinispan.configuration.cache.ClusteringConfigurationBuilder;
+import org.infinispan.configuration.cache.Configuration;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.configuration.cache.ContentTypeConfigurationBuilder;
+import org.infinispan.configuration.cache.CustomStoreConfigurationBuilder;
+import org.infinispan.configuration.cache.EncodingConfigurationBuilder;
+import org.infinispan.configuration.cache.Index;
+import org.infinispan.configuration.cache.InterceptorConfiguration;
+import org.infinispan.configuration.cache.InterceptorConfigurationBuilder;
+import org.infinispan.configuration.cache.MemoryConfigurationBuilder;
+import org.infinispan.configuration.cache.PartitionHandlingConfigurationBuilder;
+import org.infinispan.configuration.cache.SecurityConfigurationBuilder;
+import org.infinispan.configuration.cache.SingleFileStoreConfigurationBuilder;
+import org.infinispan.configuration.cache.StorageType;
+import org.infinispan.configuration.cache.StoreConfigurationBuilder;
+import org.infinispan.configuration.cache.TransactionConfiguration;
+import org.infinispan.conflict.MergePolicy;
+import org.infinispan.eviction.EvictionStrategy;
+import org.infinispan.eviction.EvictionType;
+import org.infinispan.partitionhandling.PartitionHandling;
+import org.infinispan.persistence.cluster.ClusterLoader;
+import org.infinispan.persistence.file.SingleFileStore;
+import org.infinispan.transaction.LockingMode;
+import org.infinispan.util.concurrent.IsolationLevel;
+import org.infinispan.xsite.spi.XSiteMergePolicy;
+import org.kohsuke.MetaInfServices;
+
+/**
+ * This class implements the parser for Infinispan/AS7/EAP/JDG schema files
+ *
+ * @author Tristan Tarrant
+ * @author Galder Zamarreño
+ * @since 12.0
+ */
+@MetaInfServices(ConfigurationParser.class)
+@Namespace(root = "local-cache")
+@Namespace(root = "distributed-cache")
+@Namespace(root = "replicated-cache")
+@Namespace(root = "scattered-cache")
+@Namespace(uri = NAMESPACE + "*", root = "local-cache")
+@Namespace(uri = NAMESPACE + "*", root = "distributed-cache")
+@Namespace(uri = NAMESPACE + "*", root = "replicated-cache")
+@Namespace(uri = NAMESPACE + "*", root = "scattered-cache")
+public class FragmentParser implements ConfigurationParser {
```

Comment:
How about `CacheConfigParser` or `CacheParser`?

Reply:
`CacheParser` it is. Updated.